OneCommunityGlobal/HighestGoodNetworkApp
# Masasa_onlycreatebadge_permission

# Description
This PR creates a new Permission that restrict the badge management to capability to creating badges only. Below are pictures that show the task
<img width="748" alt="Screenshot 2023-09-09 at 9 25 22 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73863365/312846d4-d360-448a-99ea-a40dd96cfceb">
<img width="1384" alt="Screenshot 2023-09-09 at 9 29 45 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73863365/9f6c0848-c8ea-4a49-b107-cfc40ac36629">


## Related PRS (if any):
This frontend PR is related to PR [#534](https://github.com/OneCommunityGlobal/HGNRest/pull/534) on the backend
…

## Main changes explained:
- Created the permision in RolesTab.jsx and named it seeBadgeManagementTab …
- In order to have  to display its info icon description I added the permission in permissionList in Rolepermissions…
-  Then used the hasPermission function in badgemanagement to determine what to display between having the both the assign and develop badge tabs…
- The same function, hasPermission was also used in badgedevelopmentTable to disable the ability to edit or delete badges
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner.. this way you can assign and unassign the permission to you designated user...
4. go to otherlinks→ Permissions Management→ manage User permisssions→ Select user…
5. assign a user you can log into the permission 'See Badge Management Tab (ONLY create Badges)'
6. log out of owner and log in as that user 
7. go to otherlinks→ Badge Management

Expect to see this if the permission has been granted succesfully.. 
<img width="1440" alt="Screenshot 2023-09-09 at 9 43 44 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73863365/5fc16ffb-f544-4668-a888-f1d91b2ef532">

 repeat and log in as owner and remove the permission and expect to see this in the badge management tab:

<img width="1346" alt="Screenshot 2023-09-09 at 9 48 54 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73863365/b47b70c1-4a6e-4f65-a9d1-5d5a071cd58b">

## Note:
Include the information the reviewers need to know.
Do not test with an owner account because the owner always has all permissions
